lib/src/widget/asset_picker_viewer.dart

```diff
@@ -109,6 +109,12 @@ class AssetPickerViewerState<Asset, Path>
     builder.initStateAndTicker(this, this);
   }
 
+  @override
+  void didUpdateWidget(covariant AssetPickerViewer<Asset, Path> oldWidget) {
+    builder.initStateAndTicker(this, this);
+    super.didUpdateWidget(oldWidget);
+  }
```

Comment:
This is incorrect for delegates overall, we should provide another method instead.

Consider adding this into the `AssetPickerViewerBuilderDelegate`:
```dart
/// Call when the viewer is calling [State.didUpdateWidget].
/// 当预览器调用 [State.didUpdateWidget] 时操作 [State] 和 [TickerProvider]。
void didUpdateViewer(
  covariant AssetPickerViewerState<Asset, Path> state,
  covariant AssetPickerViewer<Asset, Path> oldWidget,
  covariant AssetPickerViewer<Asset, Path> newWidget,
) {}
```

```suggestion
  @override
  void didUpdateWidget(covariant AssetPickerViewer<Asset, Path> oldWidget) {
    super.didUpdateWidget(oldWidget);
    builder.didUpdateViewer(this, oldWidget, widget);
  }
```

Reply:
To fix  `LateError (LateInitializationError: Field 'doubleTapAnimationController' has not been initialized.)` error, here is how i applied your suggestion :

```dart
void didUpdateViewer(
    covariant AssetPickerViewerState<Asset, Path> state,
    covariant AssetPickerViewer<Asset, Path> oldWidget,
    covariant AssetPickerViewer<Asset, Path> newWidget,
  ) {
    doubleTapAnimationController =
        oldWidget.builder.doubleTapAnimationController;
    doubleTapCurveAnimation = oldWidget.builder.doubleTapCurveAnimation;
  }
```